async/source_test.go

```diff
@@ -66,11 +68,65 @@ func TestConsumeMessagesSuccessfully(t *testing.T) {
 	}
 }
 
+func TestConsumeMessagesSuccessfullyConcurrently(t *testing.T) {
+	receivedAcks := make(chan substrate.Message)
+
+	var wg sync.WaitGroup
+
+	source := messageSourceAdapter{
+		source: &messageSourceMock{
+			consumerMessagesMock: func(ctx context.Context, messages chan<- substrate.Message, acks <-chan substrate.Message) error {
+				for i := 0; i < 100; i++ {
+					wg.Add(1)
+					messages <- message.FromString("payload")
+				}
+
+				for {
+					select {
+					case <-ctx.Done():
+						return nil
+					case ack := <-acks:
+						receivedAcks <- ack
+					}
+				}
+			},
+		},
+		consumers: 5,
+		msgBuffer: 100,
+		ackBuffer: 100,
+	}
+
+	sourceContext, sourceCancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
do we really need a timeoutable context in this test case?

Reply:
I find time out contexts allow the tests to show that the test code works within a reasonable window and don't hang.

Reply:
Simply following the pattern in the rest of the tests. I would verge on the side of leaving this, however. It means broken tests won't hang indefinitely.

Reply:
ok, ty for clarification